### PR TITLE
added filtered job executions by job name LIKE DAO method and implemented search query at API level.

### PR DESCRIPTION
**Pull request by siddhantsorann**
Related to [issue](https://github.com/spring-cloud/spring-cloud-dataflow/issues/3866)
Adding DAO method and updating SImpleJobService as well as Controller to be able to search job executions where job name is like query string coming in via the API Get Request.

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/JdbcSearchableJobExecutionDao.java [hunk 1]
```diff
@@ -207,6 +207,23 @@ public int countJobExecutions(String jobName, BatchStatus status) {
 		return getJdbcTemplate().queryForObject(getQuery(GET_COUNT_BY_JOB_NAME_AND_STATUS), Integer.class, jobName, status.name());
 	}
 
+	@Override
+	public List<JobExecutionWithStepCount> getFilteredJobExecutionsWithStepCount(String q, int start, int count) {
```
Comment (ntantri) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/JdbcSearchableJobExecutionDao.java]: Can you replace `q` with a proper name like `queryString`?

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/TaskJobService.java [hunk 2]
```diff
@@ -155,13 +155,15 @@
 	 * Retrieves Pageable list of {@link JobExecutionWithStepCount}s from the JobRepository
 	 * and matches the data with a task id but excludes the step executions.
 	 *
+	 *
+	 * @param q
```
Comment (ntantri) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/TaskJobService.java]: `q search query string to filter job names`

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/JobExecutionThinController.java [hunk 3]
```diff
@@ -86,9 +86,9 @@ public JobExecutionThinController(TaskJobService taskJobService) {
 	 */
 	@RequestMapping(value = "", method = RequestMethod.GET, produces = "application/json")
 	@ResponseStatus(HttpStatus.OK)
-	public PagedModel<JobExecutionThinResource> listJobsOnly(Pageable pageable,
-			PagedResourcesAssembler<TaskJobExecution> assembler) throws NoSuchJobExecutionException {
-		List<TaskJobExecution> jobExecutions = taskJobService.listJobExecutionsWithStepCount(pageable);
+	public PagedModel<JobExecutionThinResource> listJobsOnly(@RequestParam(required = false) String q,
```
Comment (ntantri) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/JobExecutionThinController.java]: Add the Javadoc for `@param q search query string to filter job names`

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java [hunk 4]
```diff
@@ -106,10 +107,18 @@ public DefaultTaskJobService(JobService jobService, TaskExplorer taskExplorer,
...
+		List<JobExecutionWithStepCount> jobExecutions;
+		if (StringUtils.isEmpty(q)) {
+			jobExecutions = new ArrayList<>(
+					jobService.listJobExecutionsWithStepCount(getPageOffset(pageable), pageable.getPageSize()));
+		}
+		else {
+			jobExecutions = new ArrayList<>(
+					jobService.listFilteredJobExecutionsWithStepCount(q, getPageOffset(pageable),
+							pageable.getPageSize()));
+		}
```
Comment (ntantri) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java:121]: Any UnitTests that can be added?

**Review by ntantri — CHANGES_REQUESTED** [review]

**Comment by ntantri** [comment]
Also is there any API documentation which can be updated with `q search query string to filter job names`?

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java (hunk 4 above)
Comment (siddhantsorann) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java:121]: Currently, none have been written for the previous implementation. Will try to cover this case.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/JobExecutionThinController.java (hunk 3 above)
Comment (siddhantsorann) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/JobExecutionThinController.java]: done

**Comment by siddhantsorann** [comment]
> Also is there any API documentation which can be updated with `q search query string to filter job names`?

This is being used internally, so I don't think it's part of any documentation. Will check before merging to scdf master to be sure.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java (hunk 4 above)
Comment (ntantri) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java:121]: Sure thanks.

**Pull request merged**